lib/gateway/make-resolver.js

```diff
@@ -382,13 +382,20 @@ function makeResolver ({ service, createOperation, transformData, isQuery, isRef
 
     if (isReference) {
       variables.representations = [removeNonIdProperties(parent[fieldName], type)]
-    } else if (!isQuery) {
+    } else if (!isQuery && !isSubscription) {
       variables.representations = [{
         ...removeNonIdProperties(parent, parentType),
         ...getRequiredFields(parent, schema.getType(parentType).getFields()[fieldName])
       }]
     }
 
+    if (isSubscription) {
+      const subscriptionId = service.createSubscription(query, variables)
+      const subscription = await pubsub.subscribe(`${service.name}_${subscriptionId}`)
+
+      return subscription
```

Comment:
```suggestion
      return pubsub.subscribe(`${service.name}_${subscriptionId}`)
```

We can save a few promises in this way. Also remove `async` at the top.